.github/workflows/odo-init-test.yml

```diff
@@ -0,0 +1,117 @@
+# This workflow will build and test odo-init-image and setup necessary to run integration tests
+#
+# Env required for this workflow:
+#
+# 1. Setup secrets in your repository: IBM_CLOUD_API_KEY, SSH_PRIVATE_KEY, REGISTRY_PASSWORD
+# 2. Change the values for the IC_REGION
+name: odo-init-image-test
+on: [ push ]
+
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  IBM_CLOUD_API_KEY: ${{ secrets.IBM_CLOUD_API_KEY }}
+  IBM_CLOUD_REGION: eu-de
+  OPENSHIFT_CLUSTER_ENDPOINT: ${{ secrets.OPENSHIFT_CLUSTER_ENDPOINT }}
+  KUBE_CLUSTER_ID: ${{ secrets.IBM_CLOUD_KUBE_CLUSTER_ID }}
+
+jobs:
+  build:
+    name: odo-init-image-test
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Add Cleanup Tag
+      run: echo "LABEL quay.expires-after=20h" >> Dockerfile
+
+    - name: Build Image
+      id: build-image
+      uses: redhat-actions/buildah-build@v2
+      with:
+        image: odo-init-pr
+        tags: ${{ github.sha }}
+        containerfiles: |
+          ./Dockerfile
+
+    # Podman Login action (https://github.com/redhat-actions/podman-login) also be used to log in,
+    # in which case 'username' and 'password' can be omitted.
+    - name: Push To quay.io
+      id: push-to-quay
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: ${{ steps.build-image.outputs.image }}
```

Comment:
Does this produce unique images with their own tags? If not then we are only testing using one image tag which could be a problem for PR builds

Reply:
Image name is same but with unique tag. It uses git sha as a tag.

> If not then we are only testing using one image tag which could be a problem for PR builds

Not sure what you mean by this statement, can you help me understand

Reply:
What I mean is if we are not careful with tags then we might end up with test runs interfering with each other. I guess using git sha should help.


Also I have one more question. What about tag cleanup. What are we doing to ensure that the images we push to quay are cleaned. We might end up with way too many tags otherwise as more PRs go in

Reply:
> What I mean is if we are not careful with tags then we might end up with test runs interfering with each other. I guess using git sha should help.

Yes we are using git sha as tag for the test.

> What are we doing to ensure that the images we push to quay are cleaned. We might end up with way too many tags otherwise as more PRs go in

Yeah, it's also handled in this script here [Link to comment](https://github.com/redhat-developer/odo-init-image/pull/102/files#r781007969)

Reply:
Cool so I am assuing this step will run no matter if the previous steps failed, correct?

Reply:
Which step are you referring to?

Reply:
cleanup?

Reply:
> cleanup? or we will end up leaking images

Yeah, the image cleanup will be removed from quay after 20hrs.